### PR TITLE
[k8s] Show all kubernetes clusters in optimizer table

### PR DESCRIPTION
**Pull request by Michaelvll**
Since we support multiple k8s clusters, people may expect to see multiple k8s clusters in their `allowed_contexts` that have the available GPUs.

Two clusters: 
1. my-h100-cluster-gke: 1 H100 node, 1 L4 node
2. my-h100-cluster-eks: 1 H100 node

```
sky launch --gpus H100 echo 'Hello World'
Task from command: echo Hello World
== Optimizer ==
Estimated cost: $0.0 / hour

Considered resources (1 node):
------------------------------------------------------------------------------------------------------------------------
 CLOUD        INSTANCE                       vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE           COST ($)   CHOSEN   
------------------------------------------------------------------------------------------------------------------------
 Kubernetes   2CPU--8GB--1H100               2       8         H100:1         my-h100-cluster-gke   0.00          ✔     
 Kubernetes   2CPU--8GB--1H100               2       8         H100:1         my-h100-cluster-eks   0.00                
 Lambda       gpu_1x_h100_pcie               26      200       H100:1         us-east-1             2.49                
 Cudo         sapphire-rapids-h100_1x2v4gb   2       4         H100:1         ca-montreal-3         2.83                
 Fluidstack   H100_PCIE_80GB::1              28      180       H100:1         CANADA                2.89                
 RunPod       1x_H100_SECURE                 16      80        H100:1         CA                    4.49                
 Azure        Standard_NC40ads_H100_v5       40      320       H100:1         eastus                6.98                
------------------------------------------------------------------------------------------------------------------------

Launching a new cluster 'sky-b34d-gcpuser'. Proceed? [Y/n]: 
```

```
sky launch --gpus l4
== Optimizer ==
Estimated cost: $0.0 / hour

Considered resources (1 node):
----------------------------------------------------------------------------------------------------------
 CLOUD        INSTANCE         vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE           COST ($)   CHOSEN   
----------------------------------------------------------------------------------------------------------
 Kubernetes   2CPU--8GB--1L4   2       8         L4:1           my-h100-cluster-gke   0.00          ✔     
 RunPod       1x_L4_SECURE     4       24        L4:1           CA                    0.44                
 GCP          g2-standard-4    4       16        L4:1           us-east4-a            0.70                
 AWS          g6.xlarge        4       16        L4:1           us-east-1             0.80                
----------------------------------------------------------------------------------------------------------
```

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/optimizer.py [hunk 1]
```diff
@@ -821,13 +821,14 @@ def format_number(x):
...
-                    'use_spot': resources.use_spot
-                },
-                sort_keys=True)
+            resource_key_dict = {
+                'cloud': f'{resources.cloud}',
+                'accelerators': f'{resources.accelerators}',
+                'use_spot': resources.use_spot
+            }
+            if isinstance(resources.cloud, clouds.Kubernetes):
+                resource_key_dict['region'] = resources.region
```
Comment (romilbhardwaj) [inline sky/optimizer.py:830]: Can we add a quick comment here describing why we add `region` here?

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll! Tested on GKE + local cluster, works well.

**Comment by romilbhardwaj** [comment]
Should we merge this @Michaelvll?

**Comment by Michaelvll** [comment]
Oops, missed the review. Will merge it soon

File: sky/optimizer.py (hunk 1 above)
Comment (Michaelvll) [reply sky/optimizer.py:843]: Good point! Added.

**Pull request merged**